libcperciva/util/sock_util.c

```diff
@@ -335,3 +336,36 @@ sock_addr_ensure_port(const char * addr)
 	/* Failure! */
 	return (NULL);
 }
+
+/**
+ * sock_addr_validate(addr):
+ * Check that ${addr} is syntactically valid, but do not perform any address
+ * resolution.
+ */
+int
+sock_addr_validate(const char * addr)
+{
+
+	/* Sanity checks. */
+	if (addr == NULL) {
```

Comment:
On second thought, I'm not a big fan of calling `NULL` an "empty" address.  It's not an address at all.  Also, in combination with the spiped code, forgetting to specify the address results in "Empty sock(et) address." followed by usage().

I would omit the `NULL` check from here and have the user responsible for checking for NULLity.

Reply:
... how about an `assert(addr != NULL)`?

I mean, this is shared code for a function called `sock_addr_validate()`.  Moving a sanity check out to user code seems an odd thing to do, especially since the function is only a handful of lines.

Yes, given the current design of spiped we should never get a `NULL`, but what about future use in other projects?

If you're sure you don't want even an `assert`, I'll nuke it quickly.